dipy/align/tests/test_imaffine.py

```diff
@@ -0,0 +1,251 @@
+import numpy as np
+from dipy.align import floating
+import dipy.align.vector_fields as vf
+import dipy.align.imaffine as imaffine
+import dipy.core.geometry as geometry
+from numpy.testing import (assert_array_equal,
+                           assert_array_almost_equal,
+                           assert_almost_equal,
+                           assert_equal)
+import nibabel as nib
+import dipy.align.mattes as mattes
+import scipy as sp
+from dipy.align.transforms import (Transform,
+                                   regtransforms)
+from dipy.align.imaffine import *
+import dipy.viz.regtools as rt
+import dipy.align.imaffine as imaffine
+from dipy.data import get_data
+from dipy.align.tests.test_mattes import setup_random_transform
+
+# For each transform type, select a transform factor (indicating how large the
+# true transform between static and moving images will be) and a sampling
+# (either a positive integer less than or equal to 100, or None) indicating
+# the percentage (if int) of voxels to be used for estimating the joint PDFs,
+# or dense sampling (if None)
+factors = {('TRANSLATION', 2): (2.0, 30),
+           ('ROTATION', 2): (0.1, None),
+           ('RIGID', 2): (0.1, 50),
+           ('SCALING', 2): (0.01, None),
+           ('AFFINE', 2): (0.1, 40),
+           ('TRANSLATION', 3): (2.0, None),
+           ('ROTATION', 3): (0.1, 35, 60),
+           ('RIGID', 3): (0.1, None),
+           ('SCALING', 3): (0.1, 30),
+           ('AFFINE', 3): (0.1, None)}
+
+def test_aff_centers_of_mass_3d():
+    np.random.seed(1246592)
+    shape = (64, 64, 64)
+    rm = 8
+    sp = vf.create_sphere(shape[0]//2, shape[1]//2, shape[2]//2, rm)
+    moving = np.zeros(shape)
+    # The center of mass will be (16, 16, 16), in image coordinates
+    moving[:shape[0]//2, :shape[1]//2, :shape[2]//2] = sp[...]
+
+    rs = 16
+    # The center of mass will be (32, 32, 32), in image coordinates
+    static = vf.create_sphere(shape[0], shape[1], shape[2], rs)
+
+    # Create arbitrary image-to-space transforms
+    axis = np.array([.5, 2.0, 1.5])
+    t = 0.15 #translation factor
+    trans = np.array([[1, 0, 0, -t*shape[0]],
+                      [0, 1, 0, -t*shape[1]],
+                      [0, 0, 1, -t*shape[2]],
+                      [0, 0, 0, 1]])
+    trans_inv = np.linalg.inv(trans)
+
+    for theta in [-1 * np.pi/6.0, 0.0, np.pi/5.0]: #rotation angle
```

Comment:
What's wrong with the following?

     for rotation_angle in [-1 * np.pi/6.0, 0.0, np.pi/5.0]:
          for scale_factor in [0.83,  1.3, 2.07]:
                 ...

That way, you don't need these comments, because the variable name explains what it is